fix typeguard errors

psychocoderHPC opened the pull request:
fix https://github.com/ComputationalRadiationPhysics/picongpu/issues/4853

With typeguard 3.0.0 the error code type changed from `TypeError` to `TypeCheckError`

see: https://typeguard.readthedocs.io/en/stable/versionhistory.html

BrianMarre commented:
@psychocoderHPC remember to also update the requirements.txt to typeguard>=4.2.1, since the interface change has happened somewhere between 4.1.5(last successful CI run) and now(4.2.1)

psychocoderHPC committed:
fix typeguard CI errors

With typeguard 3.0.0 the error code type changed from TypeError to TypeCheckError
Set typeguard min version to 4.2.1

see: https://typeguard.readthedocs.io/en/stable/versionhistory.html

chillenzer commented on test/python/picongpu/quick/pypicongpu/runner.py:
```diff
@@ -403,7 +403,7 @@ def test_init_sim_type(self):
 
         for invalid_sim in [None, {}, 0, ""]:
-            with self.assertRaises(TypeError):
+            with self.assertRaises(typeguard.TypeCheckError):
```
```suggestion
            with self.assertRaises(TypeError):
```
As opposed to the others, this is indeed a manually raised `TypeError` from [here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/7edbba8d21172f97cad0e6bd84fdb93098773844/lib/python/picongpu/pypicongpu/runner.py#L194). Not sure if that's desired but at least this should fix the CI.

BrianMarre commented:
superseeded by #4861, @PrometheusPi  may be closed, thanks!

ikbuibui reviewed (commented):
Closing due to #4861
